corehq/apps/user_importer/models.py

```diff
@@ -0,0 +1,26 @@
+import uuid
+
+from csv import DictWriter
+from io import StringIO
+
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+
+from corehq.blobs import CODES, get_blob_db
+
+
+class UserUploadRecord(models.Model):
+    domain = models.TextField()
+    status = JSONField(null=True)
```

Comment:
Might as well add `default=dict`. I'd prefer a set of constant/string columns for status rather than an arbitrary object, but I see `CaseUploadRecord` does the same thing so I think this is okay for consistency's sake.

Reading further, is it accurate that the entire original upload is stored in the status (in `rows`)? That seems like a lot of info to call "status", but again, if it's mimicking case uploads I won't argue about it.

Reply:
I guess its more of a result than a status, but it is just whatever is currently returned by the upload task (and that we use to populate the "you uploaded X workers, Z failed for these reasons" message to the user. I defaulted to just dumping the whole thing in, to capture everything we have rather than picking out things that seem useful now.

> is it accurate that the entire original upload is stored in the status (in rows)?
this is accurate. it contains the row itself, as well as whether or not it succeeded, and i believe the specific error if it failed.

Reply:
I think we should store user_details separately and not mix it with status. I'd read status just as failure or success and then additionally it can also be used as a filter on the admin end to filter out errored/successful uploads.

In terms of sotring what was the result of the upload, I think we should store that as a column `result` or `results`, just like what the view calls it (just the group results and user results)

nit: If user_specs are not very different from row in status_row or you have a reason to use the latter, could we just copy `user_specs` when we begin upload into this object, after removing the passwords because that would truly reflect what user actually uploaded and not what we processed it as.